src/inertial_sense_ros.cpp

```diff
@@ -5,857 +5,1227 @@
+
+            ecef_odom_msg.twist.twist.angular.x = result[0];
+            ecef_odom_msg.twist.twist.angular.y = result[1];
+            ecef_odom_msg.twist.twist.angular.z = result[2];
+
+            odom_ins_ecef_.pub.publish(ecef_odom_msg);
+        }
+
+        if (odom_ins_ned_.enabled)
+        {
+            ixVector4 qn2b;
+            ixMatrix3 Rb2n, Re2n, buf;
+
+            // NED-to-body quaternion
+            mul_Quat_ConjQuat(qn2b, qe2b, qe2n);
+            // Body-to-NED rotation matrix
+            rotMatB2R(qn2b, Rb2n);
+            // ECEF-to-NED rotation matrix
+            rotMatB2R(qe2n, buf);
+            transpose_Mat3(Re2n, buf);
+
+            // Pose
+            // Transform position from ECEF to NED and attitude from body to NED
+            transform_6x6_covariance(Pout, poseCov, Re2n, Rb2n);
+            for (int i = 0; i < 36; i++) 
+            {
+                ned_odom_msg.pose.covariance[i] = Pout[i];
+            }
+            // Twist
+            // Transform velocity from ECEF to NED and angular rate from body to NED
+            transform_6x6_covariance(Pout, twistCov, Re2n, Rb2n);
+            for (int i = 0; i < 36; i++) 
+            {
+                ned_odom_msg.twist.covariance[i] = Pout[i];
+            }
+
+            ned_odom_msg.header.stamp = ros_time_from_week_and_tow(msg->week, msg->timeOfWeek);
+            ned_odom_msg.header.frame_id = frame_id_;
+                                                                
+            // Position
+            ixVector3d llaPosRadians;
+                //ecef to lla (rad,rad,m)
+            ecef2lla(msg->ecef, llaPosRadians, 5);
+            ixVector3 ned;
+            ixVector3d refLlaRadians;
+                //convert refLla_ to radians
+            lla_Deg2Rad_d(refLlaRadians, refLla_);
+                //lla to ned
+            lla2ned_d(refLlaRadians, llaPosRadians, ned);
+
+            ned_odom_msg.pose.pose.position.x = ned[0];
+            ned_odom_msg.pose.pose.position.y = ned[1];
+            ned_odom_msg.pose.pose.position.z = ned[2];
+
+            // Attitude
+
+            ned_odom_msg.pose.pose.orientation.w = qn2b[0]; // w
+            ned_odom_msg.pose.pose.orientation.x = qn2b[1]; // x
+            ned_odom_msg.pose.pose.orientation.y = qn2b[2]; // y
+            ned_odom_msg.pose.pose.orientation.z = qn2b[3]; // z
+
+            // Linear Velocity
+            ixVector3 result, theta;
+
+            quatConjRot(result, qe2n, msg->ve);            
+
+            ned_odom_msg.twist.twist.linear.x = result[0];
+            ned_odom_msg.twist.twist.linear.y = result[1];
+            ned_odom_msg.twist.twist.linear.z = result[2];
+
+            // Angular Velocity
+
+            // Transform from body frame to NED
+            ixVector3 angVelImu = {(f_t)imu_msg.angular_velocity.x, (f_t)imu_msg.angular_velocity.y, (f_t)imu_msg.angular_velocity.z};
+            quatConjRot(result, qn2b, angVelImu);
```

Comment:
I think it needs to be quatRot() here. 
Here is its description from ISPose.c
/* Computationally simple means to apply quaternion rotation to a vector.
 * Requires quaternion be normalized first.
 * If quaternion describes current attitude, then **rotation is body -> inertial frame**.
 * Equivalent to a DCM.T * vector multiply.
 */
void quatRot(ixVector3 result, const ixQuat q, const ixVector3 v)

Reply:
OK